client/src/actions/index.ts

```diff
@@ -188,87 +189,89 @@ const dispatchDiffExpErrors = (dispatch: any, response: any) => {
+          }),
+          credentials: "include",
+        }
+      );
 
-    const res = await fetch(
-      `${globals.API.prefix}${globals.API.version}diffexp/obs`,
-      {
-        method: "POST",
-        headers: new Headers({
-          Accept: "application/json",
-          "Content-Type": "application/json",
-        }),
-        body: JSON.stringify({
-          mode: "topN",
-          count: num_genes,
-          set1: { filter: { obs: { index: set1 } } },
-          set2: { filter: { obs: { index: set2 } } },
-        }),
-        credentials: "include",
+      if (!res.ok || res.headers.get("Content-Type") !== "application/json") {
+        return dispatchDiffExpErrors(dispatch, res);
       }
-    );
 
-    if (!res.ok || res.headers.get("Content-Type") !== "application/json") {
-      return dispatchDiffExpErrors(dispatch, res);
-    }
+      const response = await res.json();
+      const varIndex = await annoMatrix.fetch("var", varIndexName);
+      const diffexpLists = { negative: [], positive: [] };
+      for (const polarity of Object.keys(diffexpLists)) {
+        // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
+        diffexpLists[polarity] = response[polarity].map((v: any) => [
+          varIndex.at(v[0], varIndexName),
+          ...v.slice(1),
+        ]);
+      }
 
-    const response = await res.json();
-    const varIndex = await annoMatrix.fetch("var", varIndexName);
-    const diffexpLists = { negative: [], positive: [] };
-    for (const polarity of Object.keys(diffexpLists)) {
-      // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-      diffexpLists[polarity] = response[polarity].map((v: any) => [
-        varIndex.at(v[0], varIndexName),
-        ...v.slice(1),
-      ]);
+      /* then send the success case action through */
+      return dispatch({
+        type: "request differential expression success",
+        data: diffexpLists,
+      });
+    } catch (error) {
+      return dispatch({
+        type: "request differential expression error",
+        error,
+      });
     }
+  };
 
-    /* then send the success case action through */
-    return dispatch({
-      type: "request differential expression success",
-      data: diffexpLists,
-    });
-  } catch (error) {
-    return dispatch({
-      type: "request differential expression error",
-      error,
-    });
-  }
-};
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any --- FIXME: disabled temporarily on migrate to TS.
-function fetchJson(pathAndQuery: any) {
```

Comment:
it would be best to make this a generic, so the caller can specify the expected return type.

eg, 
```
function fetchJson<ReturnType>(pathAndQuery: string): Promise<ReturnType> {
    ...
}
```

Ditto for doJsonRequest()